impl/asm-based-transformer/src/main/java/org/convert2ee9/Transformer.java

```diff
@@ -579,8 +570,9 @@ private static void classFile(final Transformer t, final Path source, final Path
             InputStream sourceBAIS = null;
             InputStream inputStream = Files.newInputStream(source);
             try {
-                ClassReader classReader = new ClassReader(inputStream);
-                final byte[] targetBytes = t.transform(classReader);
+                byte[] buffer = new byte[(int) source.toFile().length()];
```

Comment:
what is the max size of the byte[]?  4gb?  I don't think anyone would have a class of that size (the classloader would need a byte[ 4gb ]) but wasn't sure about streaming versus reading the entire class file.  What do you think?

Reply:
The max size of byte[] is approx. Integer.MAX_VALUE (something little bit above 2 gb.)
We will address this byte[] vs. ByteBuffer vs. InputStream issue later on the road @scottmarlow . The plan I am working on now is to unify our implementations in the way we will have just one tools code base that will be using transformer API .
This said the interface of org.wildfly.transformer.Transformer interface may change soon.

Reply:
Further more we need to discuss if org.wildfly.transformer package is the one that we want.
I am not sure.

Reply:
Note that java.lang.instrument.ClassFileTransformer is using byte[] array for class representation. Our approach (using byte[] instead of InputStream or ByteBuffer) isn't violating anything ATM I would say.

Reply:
+1 for switching to org.wildfly.transformer.

+1 for plan to unify and move forward.  I've hit Java OOM failures in the past due to communications or file handling code allocating large byte[] buffers, so thought of raising the potential large class file issue.